src/facade/dragonfly_connection.cc

```diff
@@ -1096,16 +1091,34 @@ void Connection::EnsureAsyncMemoryBudget() {
   queue_backpressure_->EnsureBelowLimit();
 }
 
+std::string Connection::LocalBindStr() const {
+  if (socket_->IsUDS())
+    return "unix-domain-socket";
+
+  auto le = socket_->LocalEndpoint();
+  return absl::StrCat(le.address().to_string(), ":", le.port());
+}
+
+std::string Connection::LocalBindAddress() const {
+  if (socket_->IsUDS())
+    return "unix-domain-socket";
+
+  auto le = socket_->LocalEndpoint();
+  return le.address().to_string();
+}
+
 std::string Connection::RemoteEndpointStr() const {
-  const bool unix_socket = socket_->IsUDS();
-  std::string connection_str = unix_socket ? "unix:" : std::string{};
+  if (socket_->IsUDS())
+    return "unix-domain-socket";
 
   auto re = socket_->RemoteEndpoint();
-  absl::StrAppend(&connection_str, re.address().to_string(), ":", re.port());
-  return connection_str;
+  return absl::StrCat(re.address().to_string(), ":", re.port());
 }
 
 std::string Connection::RemoteEndpointAddress() const {
+  if (socket_->IsUDS())
+    return "unix-domain-socket";
+
   auto re = socket_->RemoteEndpoint();
   return re.address().to_string();
 }
```

Comment:
I don't like this...What should I return? Technically is it well possible to fetch the endpoints uds file path with the same `getpeername` from `RemoteEndpoint` below. However it requires changing the return type everywhere from `boost::asio::ip::endpoint` to smth else that can hold generic strings

Reply:
I prefer this or "uds", not worth the time to improve it.